Algorithm.CSharp/LongAndShortButterflyPutStrategiesAlgorithm.cs

```diff
@@ -0,0 +1,159 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using QuantConnect.Data.Market;
+using QuantConnect.Securities.Option;
+using QuantConnect.Securities.Positions;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This algorithm demonstrate how to use OptionStrategies helper class to batch send orders for common strategies.
+    /// In this case, the algorithm tests the Butterfly Put and Short Butterfly Put strategies.
+    /// </summary>
+    public class LongAndShortButterflyPutStrategiesAlgorithm : OptionStrategyFactoryMethodsBaseAlgorithm
+    {
+        protected override int ExpectedOrdersCount { get; } = 6;
+
+        private OptionStrategy _butterflyPut;
+        private OptionStrategy _shortButterflyPut;
+
+        protected override void TradeStrategy(OptionChain chain)
+        {
+            var contractsByExpiry = chain.Where(x => x.Right == OptionRight.Put).GroupBy(x => x.Expiry);
+            foreach (var group in contractsByExpiry)
+            {
+                var expiry = group.Key;
+                var contracts = group.ToList();
+
+                if (contracts.Count < 3)
+                {
+                    continue;
+                }
+
+                var strikes = contracts.Select(x => x.Strike).OrderBy(x => x).ToArray();
+                var atmStrike = contracts.MinBy(x => Math.Abs(x.Strike - chain.Underlying.Value)).Strike;
+                var spread = Math.Min(atmStrike - strikes[0], strikes[^1] - atmStrike);
+                var itmStrike = atmStrike + spread;
+                var otmStrike = atmStrike - spread;
+
+                if (strikes.Contains(otmStrike) && strikes.Contains(itmStrike))
+                {
+                    // Ready to trade
+                    _butterflyPut = OptionStrategies.ButterflyPut(_optionSymbol, itmStrike, atmStrike, otmStrike, expiry);
+                    _shortButterflyPut = OptionStrategies.ShortButterflyPut(_optionSymbol, itmStrike, atmStrike, otmStrike, expiry);
+                    Buy(_butterflyPut, 2);
```

Comment:
Should we `break` here (`return` in python) ? Same for `LongAndShortButterflyPutStrategiesAlgorithm/LongAndShortButterflyCallStrategiesAlgorithm `